### PR TITLE
Add macroinvertebrates graphs.

### LINKED ISSUE #48: Graphing test data - Part 1 [8]
**Edit**
Begin implementing graphing functionality. For now, create test data to work with the Macroinvertebrate model.

We're aiming to make the Macro datasheet fully functional.

**Don't merge your PR until you've tested against actual StreamWebs data!**

Dependent on #44.

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes #48.

## Changes in this PR.

- [X] Add bar/pie graphs for summary data.
- [X] Add line graph for data changes over time.
- [x] Allow the user to set their own date ranges.

## Testing this PR.

1. Make sure you've run the `sites` and `macroinvertebrate` data scripts and added your site slugs
2. Run the Docker container
3. Go to `http://localhost:8000/streamwebs/site/<site_slug>/macros` (where `site_slug` is the slug of a site with macroinvertebrate data)
4. Click the radio buttons to change between line and bar/pie graphs.

### Expected Output.

A webpage with accurate line and bar/pie graphs for the test data you entered.

@osuosl/devs

### CODE REVIEW

**Comment by leian7** [comment]
Worked great when I had a single data sheet, although I did notice this thing with the pie chart: 
patcht_cut_pie
Also worked very nicely after I added two more data sheets with different dates, although the hover images of the macroinvertebrates were a little buggy upon first landing on the page (things like the hover box only having the labels and numbers with a blank space for the image, and another time having the same combo of two images in every box). It might just be something on my end though; after refreshing everything works fine.
tolerant_hover

File: streamwebs_frontend/streamwebs/models.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Site(models.Model):
     created = models.DateTimeField(auto_now_add=True)
     modified = models.DateTimeField(auto_now=True)
 
-    objects = SiteManager()
+    test_objects = SiteManager()
```
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/models.py]: Was there a reason for renaming this? Looks like you had to edit all the other datasheets to fix  the tests.

Comment (leian7) [inline streamwebs_frontend/streamwebs/models.py]: @alxngyn On line 39, 'objects' is already taken by the GeoManager.

File: streamwebs_frontend/streamwebs/views.py [hunk 2]
```diff
@@ -122,7 +132,31 @@ def user_logout(request):
...
 
 
+def graph_water(request, site_slug):
+    # TODO: actual lookup
+    return render(request, 'streamwebs/graphs/water_quality.html')
+
+
+def graph_macros(request, site_slug):
+    site = Site.objects.get(site_slug=site_slug)
+    macros = Macroinvertebrates.objects.filter(site=site)
```
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/views.py]: This needs another filter to grab ID right?

**Comment by alxngyn** [comment]
Also, I have the macro images loaded into the statics directory so we can utilize the statics feature of Django
Located here: https://github.com/osuosl/streamwebs/tree/develop/streamwebs_frontend/streamwebs/static/macro_images

File: streamwebs_frontend/streamwebs/models.py (hunk 1 above)
Comment (alxngyn) [inline streamwebs_frontend/streamwebs/models.py]: @leian7 is this something we should change for all models or just this one or change GeoManager's `objects` function?

Comment (leian7) [inline streamwebs_frontend/streamwebs/models.py]: @alxngyn Acc. to @athai, we won't need GeoManager and we should rename all the rest of the model's current ``objects`` to ``test_objects``. She says she will do it.

**Comment by LyonesGamer** [comment]
@alxngyn They're already in the statics directory, under static/streamwebs/images/macroinvertebrates/. We can't use the `{% static %}` tag anyway, since we're referencing them from a static JS file which isn't interpreted like a template.

**Comment by athai** [comment]
Looks pretty good! However, I encountered the same issue as @leian7 when it came to the rendering of the pie chart. Oddly enough, it only happens for some sites.

macro_pie

Also, the labels for the macro graph (over time) overlap each other.

macro_time

**Comment by LyonesGamer** [comment]
I'll see if I can figure out what's up with the pie chart, I haven't seen that issue but I'll see if I can reproduce it. As for the line chart, unfortunately that's what happens if you ask d3 to create a line graph with only one data point. It doesn't create the lines at all. I'm not quite sure how to handle that case.

**Comment by LyonesGamer** [comment]
@athai @leian7 Try it again, that last commit might have fixed it.

**Comment by LyonesGamer** [comment]
@athai @leian7 Okay, one more try. I think I have it this time.

**Comment by athai** [comment]
It works!
macro_sw_fixed

However, it seems that if the slivers are too small...
text_overlap

It seems fine as is, so it may not be worth fixing, but I though you'd like to know.

**Comment by LyonesGamer** [comment]
I think it's fine for now. I'm not really sure what would be the best way to fix it.

**Comment by athai** [comment]
+1

**Comment by Kennric** [comment]
@LyonesGamer - how difficult would it be to make a legend instead of labeling each pie slice? Like: http://i.stack.imgur.com/ZUvEX.jpg

**Comment by LyonesGamer** [comment]
What it should look like now:

pie

**Comment by athai** [comment]
Looks good! The legend is much nicer than labelling the slices.

**Comment by leian7** [comment]
+1. Will you end up doing something similar for the time line graph?

**Pull request merged**